flake.nix

```diff
@@ -15,6 +15,22 @@
         pkgs = import "${nixpkgs}" {
           inherit system;
           config.allowUnfree = true;
+          overlays = [
+            (final: prev: {
+              nix = prev.nix.overrideAttrs
+                (finalAttrs: prevAttrs: {
+                  patches = (prevAttrs.patches or [ ]) ++ [
+                    (final.fetchpatch {
+                      name = "include-untracked-files.patch";
+                      url = "https://github.com/NixOS/nix/compare/2.17.1...jfroche:nix:feat/include-untracked-files-2.17.1.patch";
```

Comment:
I have rebased the pull request in the Nix repository. Due to the recent changes in the main branch, specifically the transition to the libgit2 git implementation, the PR can no longer be directly applied with our current Nix version (2.17.1).
To address this, I've initiated a branch to backport the necessary changes for version 2.17.1.